test/project_types/script/layers/infrastructure/assemblyscript_task_runner_test.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+require 'project_types/script/test_helper'
+
+describe Script::Layers::Infrastructure::AssemblyScriptTaskRunner do
+  let(:ctx) { TestHelpers::FakeContext.new }
+  let(:script_id) { 'id' }
+  let(:script_name) { "foo" }
+  let(:schema) { "schema" }
+  let(:extension_point_config) do
+    {
+      "assemblyscript" => {
+        "package": "@shopify/extension-point-as-fake",
+        "version": "*",
+        "sdk-version": "*",
+      },
+    }
+  end
+  let(:extension_point_type) { "discount" }
+  let(:language) { "ts" }
+  let(:as_task_runner) { Script::Layers::Infrastructure::AssemblyScriptTaskRunner.new(ctx) }
+  let(:script_project) do
+    TestHelpers::FakeScriptProject
+      .new(language: language, extension_point_type: extension_point_type, script_name: script_name)
+  end
+
+  before do
+    Script::ScriptProject.stubs(:current).returns(script_project)
+  end
+
+  describe ".build" do
+    subject { as_task_runner.build }
+
+    it "should trigger the compilation process" do
+      File.expects(:read).with("schema")
+      File.expects(:read).with("#{script_name}.wasm")
+
+      ctx
+        .expects(:capture2e)
+        .at_most(1)
+        .returns(['output', mock(success?: true)])
+
+      subject
+    end
+
+    it "should raise error without command output on failure" do
+      output = 'error_output'
+      ctx
+        .stubs(:capture2e)
+        .returns([output, mock(success?: false)])
+
+      assert_raises(Script::Layers::Domain::Errors::ServiceFailureError, output) do
+        subject
+      end
```

Comment:
How come we don't need the lines 35-36 here?

Reply:
The exception is raised before any file operations occur, but I added an `expects().never` so hopefully thats more clear!